src/tests/Microsoft.PowerFx.Core.Tests/ExpressionTestCases/switch.txt

```diff
@@ -42,4 +42,21 @@ true
 Error({Kind:ErrorKind.Div0})
 
 >> Switch(1, 1/0, 2, 1, 3)
-Error({Kind:ErrorKind.Div0})
\ No newline at end of file
+Error({Kind:ErrorKind.Div0})
+
+// Result of switch can't be consumed.
+>> Switch(1, 1, 2, {x:4}) + 1
+Errors: Error 0-22: Invalid argument type. Expecting one of the following: Number, Text, Boolean, UntypedObject.
+
+// Expression generating void value can be used inside of the switch. If(1<0, 1, {x:1}) => void value
+>> Switch(1, 1, 2, If(1<0, 1, {x:4}))
+If(true, {test:1}, "Mismatched args (result of the expression can't be used).")
+
+>> Switch(1, 1, 1/0, 10, {x:4})
+Error({Kind:ErrorKind.Div0})
```

Comment:
Nice test. it would be good to have a similar one where an error is not returned

    >> Switch(10, 1, 1/0, 10, {x:4})
    If(true, {test:1}, "Mismatched args...")

Reply:
added.